waspc/src/Wasp/Analyzer/Parser/AST.hs

```diff
@@ -7,23 +7,28 @@ module Wasp.Analyzer.Parser.AST
   )
 where
 
+import Wasp.Analyzer.Parser.Ctx
```

Comment:
In this file I use WithCtx on a few places to enrich the AST with the information about the parsing context. This is the main change which enables to remember this context in the Parser.AST, which enables us then to use that information when creating TypeChecker.AST from it, and also to embed it into TypeErrors (which use both Parser.AST and TypeChecker.AST).